File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        std::unordered_set<int> us(nums.begin(), nums.end());
        int result = 0;
        int currentNum = 0;
        int counter = 0;

        for(const auto& n : us){
            currentNum = n;
            if(!us.contains(currentNum-1)){
                counter = 0;
                while(us.contains(currentNum++)){
                    counter++;
                }
                result = std::max(result, counter);
            }
        }

        return result;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.go
func longestConsecutive(nums []int) int {
    m := make(map[int]bool)

    for _, n := range nums{
        m[n] = true;
    }

    max := 0

    for n := range m{
        if !m[n - 1]{
            counter := 0
            i := 0
            for m[n + i]{
                counter++
                i++
            }
            if counter > max{
                max = counter
            }
        }
    }
    return max
}